contrib/insights/src/insights.cxx

```diff
@@ -0,0 +1,215 @@
+    for (auto column : columns) {
+        column.finalize();
+    }
+    return pybind11::cast(detail::data_analysis(columns));
+}
+
+}  // namespace detail
+
+void
+register_insights(py::module &m)
+{
+
+    std::vector<std::pair<const char *, double detail::column_analysis::*>> double_statistic_names = {
+        std::make_pair("numeric_mean", &detail::column_analysis::numeric_mean),
+        std::make_pair("numeric_min", &detail::column_analysis::numeric_min),
+        std::make_pair("numeric_max", &detail::column_analysis::numeric_max),
+    };
+
+    std::vector<std::pair<const char *, size_t detail::column_analysis::*>> long_statistic_names = {
+        std::make_pair("rows_seen", &detail::column_analysis::rows_seen),
+        std::make_pair("numeric_count", &detail::column_analysis::numeric_count),
+        std::make_pair("numeric_nan_count", &detail::column_analysis::numeric_nan_count),
+        std::make_pair("string_empty_count", &detail::column_analysis::string_empty_count),
+        std::make_pair("string_min_length", &detail::column_analysis::string_min_length),
+        std::make_pair("string_min_length_not_empty", &detail::column_analysis::string_min_length_not_empty),
+        std::make_pair("string_max_length", &detail::column_analysis::string_max_length),
+        std::make_pair("string_only_whitespace_count", &detail::column_analysis::string_only_whitespace_count),
+        std::make_pair("string_null_like_count", &detail::column_analysis::string_null_like_count),
+    };
+
+    std::vector<std::pair<const char *, std::string detail::column_analysis::*>> string_statistic_names = {
+        std::make_pair("example_value", &detail::column_analysis::example_value),
+    };
+
+    auto ca_class = py::class_<detail::column_analysis>(m, "ColumnAnalysis");
+    ca_class.def_readwrite("column_name", &detail::column_analysis::column_name);
+    ca_class.def("string_cardinality", [](const detail::column_analysis &ca) { return ca.string_cardinality(); });
+    ca_class.def_readwrite("string_captured_unique_values", &detail::column_analysis::string_captured_unique_values);
+    ca_class.def_readwrite("string_captured_unique_values_overflowed", &detail::column_analysis::string_captured_unique_values_overflowed);
+
+    
+    for (auto name_method_pair : long_statistic_names) {
+        auto name = name_method_pair.first;
+        auto method = name_method_pair.second;
+        ca_class.def_readwrite(name, method);
+    }
+
+    for (auto name_method_pair : double_statistic_names) {
+        auto name = name_method_pair.first;
+        auto method = name_method_pair.second;
+        ca_class.def_readwrite(name, method);
+    }
+
+    for (auto name_method_pair : string_statistic_names) {
+        auto name = name_method_pair.first;
+        auto method = name_method_pair.second;
+        ca_class.def_readwrite(name, method);
+    }
+
+    ca_class.def("to_dict", 
+        [=](const detail::column_analysis &ca) {
```

Comment:
Explicitly list the variables you want to capture by value. Otherwise you will get copies of all variables defined in the outer closure.